x/leverage/client/tests/suite.go

```diff
@@ -32,194 +37,923 @@ func (s *IntegrationTestSuite) SetupSuite() {
 	s.Require().NoError(err)
 }
 
```

Comment:
My diff is so big, github tried to collapse it.

It will probably make more sense when reviewing to look at the overall file before the diffs, due to the restructuring.